### PR TITLE
Articles for file scoped namespaces

### LINKED ISSUE #25678: File scoped namespaces
[Feature specification](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-10.0/file-scoped-namespaces.md)

A `namespace MyProject` at the top of a file would mean everything defined in that file is in that namespace.

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #25678

Update the namespace language reference for file scoped namespace declaration.
Update the programming guide on namespace declarations.

### CODE REVIEW

**Comment by BillWagner** [comment]
The snippet warnings are addressed in #25708, which isn't quite done yet. It will be ready for review later today.

I'll wait to merge this until that PR is merged, and I have a clean snippet build.

File: docs/csharp/fundamentals/types/namespaces.md [hunk 1]
```diff
@@ -26,6 +26,13 @@ Second, declaring your own namespaces can help you control the scope of class an
 
 The name of the namespace must be a valid C# [identifier name](../coding-style/identifier-names.md).
 
+Beginning with C# 10.0, you can declare a namespace for all types defined in that file, as shown in the following example:
+
+:::code language="csharp" source="snippets/namespaces/filescopednamespace.cs":::
+
+The advantage of this new syntax is that it's simpler, saving some extra braces. That makes your code easier to read.
```
Comment (Youssef1313) [inline docs/csharp/fundamentals/types/namespaces.md:33]: nit: I'd say saving horizontal space is more important than braces.

File: docs/csharp/language-reference/keywords/snippets/filescopednamespace.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+using System;
+
+namespace SampleFileScopedNamespace;
+
+class SampleClass { }
+
+interface ISampleInterface { }
+
```
Comment (IEvangelist) [inline docs/csharp/language-reference/keywords/snippets/filescopednamespace.cs:8]: What happens if I inject another `namespace` here? Does that change the remaining type defs? Do you mention this limitation or ability in anywhere?

**Review by IEvangelist — APPROVED** [review]
Love this, great job - just a question.

File: docs/csharp/language-reference/keywords/snippets/filescopednamespace.cs (hunk 2 above)
Comment (BillWagner) [reply docs/csharp/language-reference/keywords/snippets/filescopednamespace.cs:8]: I added an example to clarify those restrictions. (It's not allowed).

**Pull request merged**